ironfish-cli/src/commands/service/snapshot.ts

```diff
@@ -99,7 +110,36 @@ export default class CreateSnapshot extends IronfishCommand {
       const snapshotBaseName = path.basename(SNAPSHOT_FILE_NAME, '.tar.gz')
       const snapshotKeyName = `${snapshotBaseName}_${timestamp}.tar.gz`
 
-      const s3 = new S3Client({})
+      let s3 = new S3Client({})
+      if (flags.r2) {
+        const secret_name = 'r2-prod-access-key'
+        const client = new SecretsManagerClient({
+          region: 'us-east-1',
+        })
+        const command = new GetSecretValueCommand({ SecretId: secret_name })
+
+        this.log('Fetching secret from AWS Secrete Store.')
+
+        const response = await client.send(command)
+
+        if (response.SecretString === undefined) {
+          this.log(`Failed to fetch R2 secrete from AWS.`)
+          this.exit(1)
+        } else {
+          const secret = JSON.parse(response.SecretString) as R2Secrete
+
+          s3 = new S3Client({
+            region: 'auto',
+            endpoint: `https://a93bebf26da4c2fe205f71c896afcf89.r2.cloudflarestorage.com`,
```

Comment:
Should our account ID / endpoint be passed via a flag?

Reply:
made the endpoint value as constant. The R2 flag is on,  this S3 client will read from the constant value. I dont think its necessary for current use case to pass in the endpoint, since it only has one possible value.